src/v/cluster/shard_placement_table.cc

```diff
@@ -328,9 +328,20 @@ ss::future<result<ss::shard_id>> shard_placement_table::prepare_transfer(
         auto destination = state.target->shard;
         // check if destination is ready
         auto ec = co_await container().invoke_on(
-          destination, [&ntp, expected_log_rev](shard_placement_table& dest) {
-              auto& dest_state
-                = dest._states.emplace(ntp, placement_state{}).first->second;
+          destination,
+          [&ntp, expected_log_rev, shard_rev = state.shard_revision](
+            shard_placement_table& dest) {
+              auto dest_it = dest._states.find(ntp);
+              if (
+                dest_it == dest._states.end()
+                || dest_it->second.shard_revision < shard_rev) {
```

Comment:
are there any anomalies that could be identified by a `dest_it->second.shard_revision > shard_rev` relationship? I assumed this would be equality, but I could also see `>` having the meaning that the destination shard has at least the minimum amount of information to pass this consistency check.

Reply:
> are there any anomalies that could be identified by a `dest_it->second.shard_revision > shard_rev` relationship?

This is not that big of a deal than the `<` anomaly. The important thing is to ensure that the current state is never newer than `target` (or `assigned`), if this is not true, reconciliation can go wrong. If OTOH it is a bit newer than expected, we'll just make a shortcut in the partition path across shards. 

But this is a partial fix anyway (the main one is https://github.com/redpanda-data/redpanda/pull/17194/commits/092fa5fdfcbb94a4bd43731dad8bbdd972d51608)